partridge/geo.py

```diff
@@ -0,0 +1,38 @@
+from typing import Dict, List
+
+import pandas as pd
+
+try:
+    import geopandas as gpd
+    from shapely.geometry import LineString, Point
+except ImportError as impexc:
+    print(impexc)
+    print("You must install GeoPandas to use this module.")
+    raise
+
+
+def build_shapes(df: pd.DataFrame) -> gpd.GeoDataFrame:
+    if df.empty:
+        return gpd.GeoDataFrame(
+            {"shape_id": [], "geometry": []}, crs={"init": "EPSG:4326"}
+        )
+
+    data: Dict[str, List] = {"shape_id": [], "geometry": []}
+    for shape_id, shape in df.sort_values("shape_pt_sequence").groupby("shape_id"):
+        data["shape_id"].append(shape_id)
+        data["geometry"].append(
+            LineString(list(zip(shape.shape_pt_lon, shape.shape_pt_lat)))
+        )
+
+    return gpd.GeoDataFrame(data, crs={"init": "EPSG:4326"})
+
+
+def build_stops(df: pd.DataFrame) -> gpd.GeoDataFrame:
+    if df.empty:
+        return gpd.GeoDataFrame(df, geometry=[], crs={"init": "EPSG:4326"})
+
+    df["geometry"] = df.apply(lambda s: Point(s.stop_lon, s.stop_lat), axis=1)
+
+    df.drop(["stop_lon", "stop_lat"], axis=1, inplace=True)
+
+    return gpd.GeoDataFrame(df, crs={"init": "EPSG:4326"})
```

Comment:
I think Geopandas 0.4.0 has gotten over some minor snafus that were related to dataframe to geodataframe conversion.

That said, this minor comment:
Seems might be safer to assign the array of converted stop points (from lon/lats to shapely geom Points) to a temp var, say `temp_geoms` and then generate the gdf in the following manner:

```
gdf = GeoDataFrame(df, crs=crs, geometry=temp_geoms)
```

Reply:
Hrm I stand corrected. Just looked around and the only items are those related to having the geometry column have a different name column name than `geometry`. Perhaps it was that issue that caused me to develop a habit of leaning on the `geometry` kwarg when instantiating a GDF.

Just went back down the version of geopandas and confirmed that the pattern you used works from v1.0.0 through current:
![image](https://user-images.githubusercontent.com/6053396/50931148-a913e480-1416-11e9-8e15-fb9d4130f5b4.png)

Reply:
Ah ok. Thanks for confirming the version compatibilities!